apps/server/src/modules/roster/service/feathers-roster.service.ts

```diff
@@ -86,43 +92,170 @@ export class FeathersRosterService {
 	}
 
 	public async getUserGroups(pseudonym: string, oauth2ClientId: string): Promise<UserGroups> {
-		const courses = await this.getCourses(pseudonym, oauth2ClientId);
+		const pseudonymContext = await this.findPseudonymByPseudonym(pseudonym);
+		const user = await this.userService.findById(pseudonymContext.userId);
+
+		const externalTool = await this.validateAndGetExternalTool(oauth2ClientId);
+		const schoolExternalTool = await this.validateSchoolExternalTool(user.schoolId, externalTool.id);
+
+		const coursesUserGroups = await this.getCoursesUserGroups(pseudonymContext, schoolExternalTool);
+		const roomsUserGroups = await this.getRoomsUserGroups(pseudonymContext, schoolExternalTool);
 
-		const userGroups = {
+		const userGroups: UserGroups = {
 			data: {
-				groups: courses.map((course) => {
-					return {
-						group_id: course.id,
-						name: course.name,
-						student_count: course.students.length,
-					};
-				}),
+				groups: [...coursesUserGroups, ...roomsUserGroups],
 			},
 		};
 
 		return userGroups;
 	}
 
-	private async getCourses(pseudonym: string, oauth2ClientId: string): Promise<CourseEntity[]> {
-		const pseudonymContext = await this.findPseudonymByPseudonym(pseudonym);
-		const user = await this.userService.findById(pseudonymContext.userId);
+	public async getGroup(id: EntityId, oauth2ClientId: string): Promise<Group> {
+		let group: Group; // = { data: { students: [], teachers: [] } };
+		let room: Room;
+		try {
+			room = await this.roomService.getSingleRoom(id);
+		} catch (e) {
+			const course: CourseEntity = await this.courseService.findById(id);
+			const externalTool = await this.validateContextExternalTools(course, course.school.id, oauth2ClientId);
+			group = await this.getCourseGroup(course, externalTool);
+
+			return group;
+		}
```

Comment:
I'm not sure if it's a good idea to check the existence of the room by catching the exception. I know that's how we do it as a workaround in the Frontend. But we should think about having something like `RoomService.roomExists(id)`. Maybe checking the type of the exception would also be an improvement.